hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -564,7 +667,12 @@ def steps(self):
         return one_of(strategies)
 
     def print_start(self):
-        report(u'state = %s()' % (self.__class__.__name__,))
+        bundles_dump = []
+        for bundle_name, bundle_var_refs in self.bundles.items():
+            values = [self.names_to_values[x.name] for x in bundle_var_refs]
+            bundles_dump.append('%s=%r' % (bundle_name, values))
+        report(u'state = %s(%s)' %
+               (self.__class__.__name__, ', '.join(bundles_dump)))
```

Comment:
Hmm.

So one of the nice relatively recent changes is that rule based stateful testing now prints something pretty close to executable code. This now changes that - the thing being printed isn't really something you can use in the constructor.

How would you feel about an alternate implementation strategy that instead of adding `initializing` to `GenericStateMachine` changed `steps` so that it decided which type of rule was currently needed and emitted an initialize rule if there were any left and otherwise a normal rule? That way initialize rules could be printed by the normal step printing mechanism without any extras (also removes all the bike-shedding/documentation about `initializing` that I had above)